boa_engine/src/bytecompiler/class.rs

```diff
@@ -113,6 +113,18 @@ impl ByteCompiler<'_, '_> {
         self.emit(Opcode::GetFunction, &[index]);
         self.emit_u8(0);
 
+        let class_env = if let Some(class_name) = class.name() {
```

Comment:
Could use `class.name().filter(...)` to avoid repetition.

Reply:
I changed it to a `match` because the `filter` looked a bit awkward.